api/v1alpha1/validation.go

```diff
@@ -1224,3 +1227,244 @@ func hasPathField(rawJSON []byte) bool {
 	_, exists := data["path"]
 	return exists
 }
+
+func (a AuthProvider) Validate(ctx context.Context) []error {
+	allErrs := []error{}
+	allErrs = append(allErrs, validation.ValidateResourceName(a.Metadata.Name)...)
+	allErrs = append(allErrs, validation.ValidateLabels(a.Metadata.Labels)...)
+	allErrs = append(allErrs, validation.ValidateAnnotations(a.Metadata.Annotations)...)
+	allErrs = append(allErrs, a.Spec.Validate(ctx)...)
+
+	return allErrs
+}
+
+func (o OIDCProviderSpec) Validate(ctx context.Context) []error {
+	allErrs := []error{}
+
+	if o.Issuer == "" {
+		allErrs = append(allErrs, errors.New("issuer is required"))
```

Comment:
if wee make these errors conts we can can write tests that unwrap. Its nice for tests to explicitly test for expected errors without needing to match strings.

```suggestion
const ErrIssuerRequired errors.New("issuer is required")
[...]
		allErrs = append(allErrs, ErrIssuerRequired)
[...]		
        if errors.Is(err, ErrIssuerRequired) {
```